frontend/src/components/ReportSelect/index.tsx

```diff
@@ -110,55 +115,66 @@ export const ReportSelect = () => {
   const onExportButtonClick = (values: {
     [key: string]: { value: number | string; label: string };
   }) => {
-    const baseUri = `https://localhost:8080/report/projects`;
     const reportUri = `${values.report_type}`;
-    let url = "";
-    // This will need to change as more report types are added
-    if (values.project) {
-      url = `${baseUri}/${values.project.value}/${reportUri}`;
-    } else {
-      url = `${baseUri}/${reportUri}`;
-    }
-    // Build querystring params
+    let url = `report/projects/${reportUri}`;
+    let routeParam;
     const querystringParams = new URLSearchParams();
-    for (const field in values) {
-      if (queryFields.includes(field)) {
-        if (values[field] instanceof Array) {
-          const options = values[field] as unknown as IOption[];
-          for (const value of options) {
-            querystringParams.append(field, value.value.toString());
+    try {
+      // Build querystring and route params from input values.
+      if (reportParamCategory) {
+        for (const param of reportParamCategory) {
+          if (values[param.field.fieldName]) {
+            const field = param.field.fieldName;
+            const value = values[param.field.fieldName];
+            // If the input is query type, add the value(s) to the querystring.
+            if (param.type === requestTypes.query) {
+              if (value instanceof Array) {
+                // Multiselect input values come as arrays and each element must be added to querystring.
+                const options = value as unknown as IOption[];
+                for (const value of options) {
+                  querystringParams.append(field, value.value.toString());
+                }
+              } else {
+                querystringParams.append(field, values[field].value.toString());
+              }
+            } else {
+              // If the request type is route, we will add that value to the route params.
+              routeParam = value;
+            }
+          } else if (param.isRequired) {
+            throw new Error(`${param.field.fieldName} field is required.`);
           }
-        } else {
-          querystringParams.append(field, values[field].value.toString());
         }
       }
+      // This will need to change as more report types are added
+      if (routeParam) {
+        url = `report/projects/${routeParam}/${reportUri}`;
```

Comment:
this seems a bit weird that you change the url where the routeParam get injected before the reportUri
Also i feel like this should have a function call to use https://github.com/bcgov/gdx-agreements-tracker/blob/development/frontend/src/utils/apiAxios.ts#L7 to create an absolute url.